base/src/main/java/com/smartdevicelink/managers/screen/BaseTextAndGraphicManager.java

```diff
@@ -154,821 +145,363 @@ public void dispose(){
-		}
-
-		if (title != null && shouldUpdateTitleField()) {
-			show.setTemplateTitle(title);
-		}
-
-		List<String> nonNullFields = findValidMainTextFields();
-		if (nonNullFields.isEmpty()){
-			return show;
-		}
-
-		int numberOfLines = defaultMainWindowCapability != null ? ManagerUtility.WindowCapabilityUtility.getMaxNumberOfMainFieldLines(defaultMainWindowCapability) : 4;
-
-		switch (numberOfLines) {
-			case 1: show = assembleOneLineShowText(show, nonNullFields);
-				break;
-			case 2: show = assembleTwoLineShowText(show);
-				break;
-			case 3: show = assembleThreeLineShowText(show);
-				break;
-			case 4: show = assembleFourLineShowText(show);
-				break;
-		}
-
-		return show;
-	}
-
-	private Show assembleOneLineShowText(Show show, List<String> showFields){
-
-		StringBuilder showString1 = new StringBuilder();
-		for (int i = 0; i < showFields.size(); i++) {
-			if (i > 0) {
-				showString1.append(" - ").append(showFields.get(i));
-			}else{
-				showString1.append(showFields.get(i));
-			}
-		}
-		show.setMainField1(showString1.toString());
-
-		MetadataTags tags = new MetadataTags();
-		tags.setMainField1(findNonNullMetadataFields());
-
-		show.setMetadataTags(tags);
-
-		return show;
-	}
-
-	private Show assembleTwoLineShowText(Show show){
-
-		StringBuilder tempString = new StringBuilder();
-		MetadataTags tags = new MetadataTags();
-
-		if (textField1 != null && textField1.length() > 0) {
-			tempString.append(textField1);
-			if (textField1Type != null){
-				tags.setMainField1(textField1Type);
+	private synchronized void sdlUpdate(final CompletionListener listener) {
+		if (this.transactionQueue.getTasksAsList().size() > 0) {
+			// Transactions already in queue, we need to clear it out
+			transactionQueue.clear();
+			updateOperation = null;
+			if (currentOperationListener != null) {
+				currentOperationListener.onComplete(false);
 			}
 		}
 
-		if (textField2 != null && textField2.length() > 0) {
-			if (( textField3 == null || !(textField3.length() > 0)) && (textField4 == null || !(textField4.length() > 0))){
-				// text does not exist in slots 3 or 4, put text2 in slot 2
-				show.setMainField2(textField2);
-				if (textField2Type != null){
-					tags.setMainField2(textField2Type);
-				}
-			} else if (textField1 != null && textField1.length() > 0) {
-				// If text 1 exists, put it in slot 1 formatted
-				tempString.append(" - ").append(textField2);
-				if (textField2Type != null){
-					List<MetadataType> typeList = new ArrayList<>();
-					typeList.add(textField2Type);
-					if (textField1Type != null){
-						typeList.add(textField1Type);
-					}
-					tags.setMainField1(typeList);
-				}
-			}else {
-				// If text 1 does not exist, put it in slot 1 unformatted
-				tempString.append(textField2);
-				if (textField2Type != null){
-					tags.setMainField1(textField2Type);
-				}
+		// Task can be READY, about to start and popped of the queue, so we have to cancel it, to prevent it from starting
```

Comment:
Do you think we combine the two if statements in one?

```
// Task can be READY, about to start and popped of the queue, so we have to cancel it, to prevent it from starting
// Or if Task is IN_PROGRESS it's not on the queue, we need to cancel it, operation will take care of stopping it.
if (updateOperation != null && (updateOperation.getState() == Task.READY || updateOperation.getState() == Task.IN_PROGRESS)) {
	updateOperation.cancelTask();
	if (currentOperationListener != null) {
		currentOperationListener.onComplete(false);
	}
}
```